### PR TITLE
 Bulk insert of failed receipts using query macro 

### LINKED ISSUE #273: [Feat.Req] Bulk insert of failed receipts
**Problem statement**
We have a loop that inserts all the failed receipts, but it's way more efficient if we insert via bulk.

### PR DESCRIPTION
**Pull request by taslimmuhammed**
Fixed #273 
Added error logs storing to the table
Successfully ran commands:-
cargo clippy --all-targets --all-features 
cargo fmt 
cargo sqlx prepare --workspace -- --all-targets
please do recheck on your device

### CODE REVIEW

File: migrations/20230915230734_tap_ravs.down.sql [hunk 1]
```diff
@@ -1,2 +1,2 @@
-DROP TABLE IF EXISTS scalar_tap_ravs CASCADE;
-DROP TABLE IF EXISTS scalar_tap_rav_requests_failed CASCADE;
+-- DROP TABLE IF EXISTS scalar_tap_ravs CASCADE;
+-- DROP TABLE IF EXISTS scalar_tap_rav_requests_failed CASCADE;
```
Comment (gusinacio) [inline migrations/20230915230734_tap_ravs.down.sql:2]: Is there any reason to remove this migration? If not, could you add it back?

File: tap-agent/src/agent/sender_allocation.rs [hunk 2]
```diff
@@ -696,31 +705,53 @@ impl SenderAllocationState {
...
+            &allocation_ids,
+            &timestamps,
+            &nounces,
+            &values,
+            &error_logs
+        )
+        .execute(&self.pgpool)
+        .await
+        .map_err(|e| {
+            error!("Failed to store receipt: {}", e);
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_allocation.rs:751]: Could you update this error to something like `"Failed to store invalid receipt: {}"`

**Review by gusinacio — CHANGES_REQUESTED** [review]
I think this is the last review. It has been a great work. Looking forward to fix these comments and merge it.

One suggestion: don't close and open new PRs unless they are have totally different code from each other. I see that you created PRs #326, #327, #328 and now #329 but they are all pointing to the same branch with one extra commit in each of them. It's hard to keep track of suggestions in other PRs.

File: migrations/20230915230734_tap_ravs.down.sql (hunk 1 above)
Comment (taslimmuhammed) [reply migrations/20230915230734_tap_ravs.down.sql:2]: Sorry for that, failed to notice while sending pr, I'll update asap

File: tap-agent/src/agent/sender_allocation.rs (hunk 2 above)
Comment (taslimmuhammed) [reply tap-agent/src/agent/sender_allocation.rs:751]: sure

**Comment by taslimmuhammed** [comment]
Please take a look at the latest commit, hope everything's good

**Review by gusinacio — APPROVED** [review]
This all looks good to me. Could you do a rebase on your side to update the commit messages so it passes our CI? This is a [guide](https://github.com/graphprotocol/indexer-rs/blob/main/CONTRIBUTORS.md) on how to do your commit messages.

**Review by carlosvdr — APPROVED** [review]

**Pull request merged**